src/components/SingleArticle.jsx

```diff
@@ -6,32 +6,82 @@ import * as api from '../api'
 export default function SingleArticle () {
     const [article, selectArticle] = useState([])
     const [isLoading, setIsLoading] = useState(true)
-
+    const [addVotes, setAddVotes] = useState(false)
+    const [subVotes, setSubVotes] = useState(false)
     const { article_id } = useParams()
 
     useEffect(() => {
         setIsLoading(true)
         api.fetchSingleArticle(article_id).then(({data}) => {
-            selectArticle(data.article.article[0])
+            selectArticle(data.article.article)
             setIsLoading(false)
         })
     }, [article_id]
     )
 
+    const upVote = (article_id) => {
+        if (addVotes === false) {
+            setAddVotes(true)
+            selectArticle((article) => {
+                return article.map((votes) => {
+                    return { ...votes, votes: article[0].votes + 1 }
+                })
+            })
+            api.incVote(article_id).catch(() => {
+                selectArticle((article) => {
+                    return article.map((votes) => {
+                        return { ...votes, votes: article[0].votes - 1 }
```

Comment:
Could you just add one small thing and just have a little message appear to say that something has gone wrong, and to try again?
Just for a better user experience